src/effects/backends/builtin/whitenoiseeffect.cpp

```diff
@@ -50,24 +58,41 @@ void WhiteNoiseEffect::processChannel(
 
     WhiteNoiseGroupState& gs = *pState;
 
-    CSAMPLE drywet = static_cast<CSAMPLE>(m_pDryWetParameter->value());
+    const CSAMPLE drywet = static_cast<CSAMPLE>(m_pDryWetParameter->value());
     RampingValue<CSAMPLE_GAIN> drywet_ramping_value(
             drywet, gs.previous_drywet, engineParameters.framesPerBuffer());
+    auto fade_in_out_ramping = RampingValue<CSAMPLE_GAIN>(
+            enableState == EffectEnableState::Enabling ? 0 : 1,
+            enableState == EffectEnableState::Disabling ? 0 : 1,
+            engineParameters.framesPerBuffer());
 
-    std::uniform_real_distribution<> r_distributor(0.0, 1.0);
+    // Is this relying on implementation details?
+    DEBUG_ASSERT(engineParameters.samplesPerBuffer() ==
+            engineParameters.framesPerBuffer() *
+                    engineParameters.channelCount());
 
-    for (unsigned int i = 0; i < engineParameters.samplesPerBuffer(); i++) {
-        CSAMPLE_GAIN drywet_ramped = drywet_ramping_value.getNext();
+    for (unsigned int frameIndex = 0;
+            frameIndex < engineParameters.framesPerBuffer();
+            frameIndex++) {
+        const CSAMPLE_GAIN drywet_ramped = drywet_ramping_value.getNext();
+        const CSAMPLE_GAIN fade_in_out_gain = fade_in_out_ramping.getNext();
 
-        float noise = static_cast<float>(
-                r_distributor(gs.gen));
+        for (unsigned int sampleInFrameIndex = 0;
+                sampleInFrameIndex < engineParameters.channelCount();
+                sampleInFrameIndex++) {
+            constexpr float normalization_divisor =
+                    static_cast<float>(std::numeric_limits<int32_t>::max());
+            gs.random_state = nextState(gs.random_state);
+            const auto bipolar_random = static_cast<int32_t>(gs.random_state);
+            const float noise = static_cast<float>(bipolar_random) / normalization_divisor;
```

Comment:
This is the final CSAMPLE
```suggestion
            const CSAMPLE noise = static_cast<float>(bipolar_random) / normalization_divisor;
```